### ISSUE TITLE
Entombed cards added twice when revealed during Joust

### ISSUE THREAD

**Issue by AnonymousUser52** [HearthSim/Hearthstone-Deck-Tracker#2937]
I was playing a Shaman deck against a Priest, and I played Hallazeal. My opponent Entombed my Hallazeal, and it showed up in their decklist tracker grayed out, with a little gift-box icon next to it, to show that it was a stolen card in their deck.

Later, they used a Healing Wave from a Shifting Shade, and it revealed my stolen Hallazeal from their deck. Then, to my surprise, their decklist tracker added another Hallazeal to their decklist, this time without the gift-box icon, to show that they had a Hallazeal in their deck that THEY OWNED. It was placed above the one that symbolized mine. That was obviously impossible, because Priest cannot put Hallazeal in their deck.

In conclusion, there is clearly an issue in the code that deals with Jousts, where it forgets to account for stolen copies of your cards put in the opponent's deck (currently only from being Entombed, but future card releases could change that), before adding that card to your opponent's decklist.

This isn't a hugely common bug, considering that Control Priest is very rare, and Joust cards are very rare, and so Joust cards revealing Entombed minions is extremely rare. It isn't even that significant, being a minor graphical glitch, but it is there, and it would be nice if it was fixed.

**Comment by jleclanche** [comment]
I don't think it's a bug - it's actually intended by blizzard, given that jousts create a *copy* of the minion, it's impossible to know whether the jousted minion is a different one or the same one.

That said, HDT could simply not add a second copy of the minion if one is already present as it can't know better.

**Comment by AnonymousUser52** [comment]
@jleclanche It creates a copy of the minion, instead of reading the minion itself? That seems like a strange programming choice, but I hope (not necessarily assume) they have their reasons. Well, if that's what they're doing, I suppose this is even less relevant than I thought.

Although, I don't think it would be TOO complicated to add a bit of code that checks if the card shown in the Joust can be in that person's deck (Hallazeal in Priest could not be a thing).

**Comment by jleclanche** [comment]
Yes, it's meant to prevent info leaks (eg. knowing when the opponent draws a card that was jousted).

**Comment by jleclanche** [comment]
More info: https://github.com/HearthSim/hs-bugs/issues/132
